docs/root/operations/cli.rst

```diff
@@ -126,6 +130,11 @@ following are the command line options that Envoy supports.
    :%z:	ISO 8601 offset from UTC in timezone ([+/-]HH:MM) ("-07:00")
    :%%:	The % sign ("%")
 
+.. option:: --log-format-escape-newlines
+
+  *(optional)* This flag enables application log sanitization to escape newline characters.
+  This prevents a single log line from spanning multiple lines in the underlying log.
```

Comment:
Would there ever be a case for more general escaping, e.g. NUL characters? I'm thinking this is unlikely, as logs are already supposed to be text format, but /shrug.

Reply:
I don't think so. We just escape newlines to ensure that logs are structured. Once they are, it's fairly easy for log processors to escape any other characters on a line-by-line basis.

Reply:
Actually I take this back. It might be cleaner just to use absl::CEscape to escape all c-style escape sequences. This will also allow our logger to support structured JSON formatting (`"` and `'` will be escaped)